docs/en/week11/11-2.md

```diff
@@ -0,0 +1,382 @@
+
+The illustration below shows how Beam Search can lead to a better sequence.
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/bs3.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 2:</b> Beam Search : Stage 1 <br>
+<br>
+</center>
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/bs1.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 3:</b> Beam Search : Stage 2<br>
+<br>
+</center>
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/bs2.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 4:</b> Beam Search : Stage 3<br>
+<br>
+</center>
+
+
+## Graph Transformer Networks
+
+We have previously seen Weighted Finite State Automata (WFSA) being used to represent the alignment graphs, as shown before.
+
+
+Graph Transformer Networks (GTNs) are basically WSFA with automatic differentiation.
+
+Lets look at key differences between Neural Networks (NNs) and GTNs
+
+
+
+|  | NN | GTN |
+| -------- | -------- | -------- |
+| Core Data Structure     | Tensor     | Graph/WFSA     |
+|     | Matrix Multiplcation     | Compose     |
+| Core Operations  | Reduction operations (Sum, Product)| Shortest Distance (Viterbi,Forward) |
+| | Negate, Add, Subtract, ...     | Closure, Union, Concatenate, ...    |
+
+
+## Why Differentiable WSFAs?
+
+* Encode Priors: easy to encode priors into a graph
+* End-to-end: Bridge the training and test process 
+* Facilitate research: When we separate the data from the code it makes it easier to explore different ideas. In our case graph is the data and the code is operations allows us to explore different ideas by changing the graph.
+
+
+## Sequence Criteria with WFSAs
+
+It turns out that lots of the sequence criteria like connection temporal classification (CTC) can be specified as the difference between two forward scores.
+
+* Graphs A: function of input X (speech) and target "output" Y (transription).
+* Graph Z: function of input X, which servers as normalization.
+* Loss: $$logP(Y \mid X) = forwardScore(A_{X,Y}) - forwardScore(Z_{X})$$
+
+Other common loss functions in ASR:
+
+* Automatic Segmentations Criterion (ASG)
+* Connectionist TEmporal  Classification (CTC)
+* Lattice Free MMI (LF-MMI)
+
+<!-- <center> -->
+
+### Lines of code for CTC: Custom vs GTN
+
+|            | GTN   |
+| ---------- | ----- |
+| Warp-CTC   | 9,742 |
+| wav2letter | 2,859 |
+| PyTorch    | 1,161 |
+| GTN        | 30*   |
+
+\*Same graph code works for decoding.
+
+<!-- </center> -->
+
+
+## Different Types of WFS
+
+### Weighted Finite-State Acceptor (WFSA)
+
+The bold circle is the starting state and the accepting state is the one with concentric circles. 
+
+#### Example
+
+A simple WFSA which recognizes aa or ab
+* The score of **aa** is 0 + 2 = 2
+* The score of **ba** is 1 + 2 = 3
+
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/figure1.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 1:</b> WFSA<br>
```

Comment:
This doesn't seem to be figure 1…
Also, there's no caption. Only a cryptic WFSA. Add an actual caption, description of the image. Here and elsewhere.